server/app/modules/MainModule.java

```diff
@@ -84,4 +90,15 @@ public OidcClientProviderParams provideOidcClientProviderParams(
     return OidcClientProviderParams.create(
         config, profileFactory, idTokensFactory, accountRepositoryProvider);
   }
+
+  @Provides
+  public ProgramSlugHandler provideProgramSlugHandler(
+      HttpExecutionContext httpContext,
+      ApplicantService applicantService,
+      ProfileUtils profileUtils,
+      ProgramService programService,
+      LanguageUtils languageUtils) {
+    return new ProgramSlugHandler(
+        httpContext, applicantService, profileUtils, programService, languageUtils);
+  }
```

Comment:
**question (blocking):** most injected classes don't need a provider. Why is this needed? Is there an injection error without it?

Reply:
TIL, thanks. Removed.